PipelinesTasks/PRMetrics/buildAndReleaseTask/invokers/azureReposInvoker.ts

```diff
@@ -0,0 +1,219 @@
+      */
+    public setCommentThreadStatus = async (commentThreadId: number, status: CommentThreadStatus): Promise<void> => {
+        this.taskLibWrapper.debug('* AzureReposInvoker.setCommentThreadStatus()');
+
+        const gitApi = await this.openConnection();
+        const updatedCommentThread: GitPullRequestCommentThread = {};
+        updatedCommentThread.status = status;
+        await gitApi.updateThread(updatedCommentThread, this.repositoryId, this.pullRequestId, commentThreadId, this.project);
+    }
+
+    /**
+      * Creates new comment thread.
+      * @param comment Comment text.
+      * @param fileName File name to be used in the comment.
+      * @param withLinesAdded Flag to determine if lines added or not.
+      */
+    public createCommentThread = async (comment: string, fileName: string, withLinesAdded: boolean): Promise<GitPullRequestCommentThread> => {
+        this.taskLibWrapper.debug('* AzureReposInvoker.createCommentThread()');
+
+        const gitApi = await this.openConnection();
+        const updatedCommentThread: GitPullRequestCommentThread = {
+            comments: [{ content: comment }],
+            threadContext: { filePath: `/${fileName}` }
+        };
+
+        const fileStart = {
+            line: 1,
+            offset: 1
+        };
+
+        const fileEnd = {
+            line: 1,
+            offset: 2
+        };
+
+        if (fileName) {
+            if (!withLinesAdded) {
+                updatedCommentThread.threadContext = {
+                    filePath: `/${fileName}`,
+                    leftFileStart: fileStart,
+                    leftFileEnd: fileEnd
+                };
+            } else {
+                updatedCommentThread.threadContext = {
+                    filePath: `/${fileName}`,
+                    rightFileStart: fileStart,
+                    rightFileEnd: fileEnd
+                };
+            }
+        }
+
+        return await gitApi.createThread(updatedCommentThread, this.repositoryId, this.pullRequestId, this.project);
+    }
+
+    /**
+      * Creates new comment.
+      * @param commentThreadId Comment thread id to add the comment.
+      * @param parentCommentId Parent comment id.
+      * @param commentText Text of the new comment.
+      */
+    public createComment = async (commentThreadId: number, parentCommentId: number, commentText: string): Promise<void> => {
+        this.taskLibWrapper.debug('* AzureReposInvoker.createComment()');
+
+        const gitApi = await this.openConnection();
+        const comment: Comment = { content: commentText, parentCommentId: parentCommentId };
+
+        await gitApi.createComment(comment, this.repositoryId, this.pullRequestId, commentThreadId, this.project);
+    }
+
+    /**
+      * Adds metadata to the pull request.
+      * @param metadata Metadata object.
+      */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
```

Comment:
Remove this suppression

Reply:
To be able to use it we need to define an interface with the possible values and use that as type instead of any. I'll see if we can do that, otherwise it has to stay as any.